sql/partitioning/partition_base.h

```diff
@@ -0,0 +1,1034 @@
+#ifndef PARTITION_BASE_INCLUDED
+#define PARTITION_BASE_INCLUDED
+
+/*
+   Copyright (c) 2005, 2017, Oracle and/or its affiliates. All rights reserved.
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; version 2 of the License.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301  USA */
+
+#include "item.h"                           // enum_monotonicity_info
+#include "partitioning/partition_handler.h" /* Partition_handler */
+
+namespace native_part {
+
+/** class where to save partitions Handler_share's */
+class Parts_share_refs {
+ public:
+  uint num_parts;            /**< Size of ha_share array */
+  Handler_share **ha_shares; /**< Storage for each part */
+  Parts_share_refs();
+  ~Parts_share_refs();
+  bool init(uint arg_num_parts);
+};
+
+/**
+  Partition specific Handler_share.
+*/
+class Partition_base_share : public Partition_share {
+ public:
+  /** Storage for each partitions Handler_share */
+  Parts_share_refs *partitions_share_refs;
+  Partition_base_share();
+  ~Partition_base_share();
+  bool init(uint num_parts);
+};
+
+class Partition_base : public handler,
```

Comment:
Yes, it's a good idea. I found the way how to make it automatically.

```
 run-clang-tidy-3.8.py -header-filter='partition_base.h' -checks='-*,modernize-use-override' -fix ./sql/partitioning/partition_base.cc -p ../ps-build-8.0
```

ps-build-8.0 must contain build commands file generated by cmake with CMAKE_EXPORT_COMPILE_COMMANDS option.

BTW, it can also do other useful things to convert a code to c++11:
clang-tidy --list-checks -checks='*' | grep "modernize"
    modernize-loop-convert
    modernize-make-unique
    modernize-pass-by-value
    modernize-redundant-void-arg
    modernize-replace-auto-ptr
    modernize-shrink-to-fit
    modernize-use-auto
    modernize-use-default
    modernize-use-nullptr
    modernize-use-override